### DISCUSSION TITLE
Nodemailer emails not working in production, 504 timeout error

### DISCUSSION THREAD

**Discussion by ashiwanikumar** [vercel/community#2137] [Help]
### Summary

Hi Vercel Team,

We have setup nodemailer in the app to send contact us page emails. It works in localhost development, however, when run on production, the emails are not being sent and it is being timed out.

We are currently on a Hobby plan. Could you please tell us why this is happening?

**Comment by romanalbella** [comment]
Hi @ashiwanikumar same thing on me. Still having the issue or already fixed?
Thanks!

**Reply by ashiwanikumar** [reply]
No still we are facing issues.

**Reply by romanalbella** [reply]
I have deployed my app to Railway and Fly.io and it works fine! Maybe the problem must be something related to Vercel himself.
But honestly I prefer to deploy in vercel. So I'm still waiting for it to be fixed.

**Comment by amyegan** [comment]
I don't have the details of your app so I don't have specific suggestions, but I can share some general info related to sending email via serverless functions. Hobby accounts have a 10 second [Serverless Function Execution Timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout), which may be the cause if the function is taking longer. Some other common problems with nodemailer are [missing environment variables](https://stackoverflow.com/a/70782245) and [not awaiting `sendMail`](https://stackoverflow.com/a/74741573). 

In many cases, it is worth using a third party email service since they can also give you analytics, scheduling, and other beneficial features. You can find a few mail provider options and more info about sending Emails from an application on Vercel here: https://vercel.com/guides/sending-emails-from-an-application-on-vercel
